packages/ag-charts-community/src/chart/data/processors.ts

```diff
@@ -297,8 +297,7 @@ export function diff(
                 if (added.has(prevId)) {
                     if (updateMovedDatums || !arraysEqual(added.get(prevId).values, prev.values)) {
                         updated.set(prevId, prev);
-                        const updatedIndex = added.get(prevId);
-                        updatedIndices.set(prevId, updatedIndex);
+                        updatedIndices.set(prevId, i);
```

Comment:
While waterfall animations are pending, reverting this fixes an issue with the pie sectors which expect the other index here https://github.com/ag-grid/ag-charts/blob/latest/packages/ag-charts-community/src/chart/series/polar/pieSeries.ts#L1772-L1774

We should re-evaluate tracking both the before and after index later.